### ISSUE TITLE
Problems adding custom metrics into models

### ISSUE THREAD

**Issue by KB261999** [plaidml/plaidml#1897]
Hey people ! 
I am trying to add a custom F1 score to VGG16 model. When the code is running and the model is compiling, the code keeps executing codes within the plaidml and doesn't move on with the training, I'm not sure what is going on exactly.
Keras==2.2.4
tensorflow==2.2
tensorflow-gpu==2.2
The code:
```
def load_CNN_model():
    base_model = VGG16(include_top=False,
    weights='imagenet',
    input_shape=(number_of_channels, 350,3),
    classes=2)
    for layer in base_model.layers:
        layer.trainable = True
    last_layer = base_model.get_layer('block5_pool')
    last_output = last_layer.output
    x = GlobalAveragePooling2D()(last_output)
    x = Dense(units=512, activation='relu')(x)
    x = Dense(units=512, activation='relu')(x)
    x = Dropout(0.2)(x)
    x = Dense(units=1, activation='sigmoid')(x)
    model = Model(base_model.input, x)
    model.compile(loss='binary_crossentropy',
              optimizer=Adam(lr=0.001),
              metrics=['accuracy', F1_scores],
              run_eagerly=True)
    return model
```
Where the metric function is:
```
def get_parameters(y_true, y_pred):
    true_positives = 0
    false_negatives = 0
    false_positives = 0
    true_negatives = 0
    for i, predict in enumerate(y_pred):
        prediction = predict[0]
        true_value = y_true[i][0]
        if true_value == 1 and prediction > 0.5:
            true_positives += 1
        elif prediction > 0.5 and true_value == 0:
            false_positives += 1
        elif prediction <= 0.5 and true_value == 1:
            false_negatives += 1
        elif prediction <= 0.5 and true_value == 0:
            true_negatives += 1
    return true_positives, false_positives, false_negatives, true_negatives

def F1_scores(y_true,y_pred):
    true_positives, false_negatives, false_positives, true_negatives = get_parameters(y_true,y_pred)
    f1_score = true_positives/(true_positives+0.5*(false_positives+false_negatives))
    return f1_score
```
Is there a function I can use to cast the metric input into a numpy array or a tensor (initially I used y_pred.numpy() but the object wasn't tensor and didn't have this function), if not, what can I do to avoid this issue ? Or is this a bug ?
